packages/api/lib/rulesHelpers.js

```diff
@@ -302,7 +307,7 @@ async function deleteRuleResources(knex, rule) {
     break;
   }
   case 'sns': {
-    if (rule.state === 'ENABLED') {
+    if (rule.rule.arn) {
```

Comment:
What's the idea behind this change?

Reply:
I see it takes the logic previously in `deleteOldEventSourceMappings`. The ENABLED check is no longer relevant?

Reply:
Let me revisit that in test for a short bit.     re-looking at this, I believe originally that was intended to logic-gate removal of triggers that shouldn't exist if the rule isn't enabled. 

Excellent question.

Reply:
@npauzenga  so in the case of *sns* rules, disabling the rule/creating the rule without an ARN deletes the ARN and disables the subscription, so in the case of API writes/updates for SNS rules, there's no ARN in the case of a rule that's disabled via business logic expectations.     I believe the original logic was just using the other corner of this logic, however an ARN being present seems like a more reliable indicator that a subscription should be removed.  

There's a good argument that we should update deleteSnsTrigger to handle a NotFoundException 

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sns/classes/unsubscribecommand.html

🤔

Reply:
Actually there isn't, as I believe the call to `unsubscribe` is idempotent.      I've added a test to encapsulate the logic in the helper, but am open to feedback: b7cbb35c76

Reply:
I also went out and manually validated the expected behavior from the endpoint for sanity reasons.

Reply:
Great, thanks @Jkovarik. That makes sense to me and the test looks good. Thanks for validating.